### ISSUE TITLE
Composer issues | key:generate | lampp webroot ?

### ISSUE THREAD

**Issue by ncepanec** [snipe/snipe-it#3965]
Hello, 

I am a Linux rookie so please go easy on me...
I'm currently trying to set up SnipeIT using CentOS7.
I used this *https://www.vultr.com/docs/how-to-install-snipe-it-on-centos-7*
  tutorial and hopefully i've solved everything until step 6 just by 
installing Xampp. I've set up the database and got apache running. 
Now before I installed composer I've deleted everything in /opt/lampp/htdocs/* 
because I read that that is the web root folder when using Xampp. After that I cloned 
SnipeIT from git in a new folder meaning I have /opt/lampp/htdocs/snipe-it/->HERE<-

Composer version = 1.5.2

Additional commands I've used after step 6: 

sudo ln -s /opt/lampp/bin/php /usr/local/bin/php        ( linking composer to work globally ? ) 

I set the all the permissions as it's mentioned in the tutorial but i get stuck while trying to use composer.

"Install PHP dependencies using Composer.

sudo composer install --no-dev --prefer-source"
-> 
sudo: composer: command not found

This appears from which ever folder I try to run this from, I've tried moving composer and restarted
my VM many times and just can't seem to find the solution. Got it working once without using Lampp 
but it seemed like this would be easier. 

If I try to use

php artisan key:generate for the key in the .env file  which I've previously configured this is what I get  : 

Warning: require(/opt/lampp/htdocs/snipe-it/bootstrap/../vendor/autoload.php): failed to open stream: No such file or directory in /opt/lampp/htdocs/snipe-it/bootstrap/autoload.php on line 17

Fatal error: require(): Failed opening required '/opt/lampp/htdocs/snipe-it/bootstrap/../vendor/autoload.php' (include_path='.:/opt/lampp/lib/php') in /opt/lampp/htdocs/snipe-it/bootstrap/autoload.php on line 17

I've tried using the common solutions and probably checked every page related to this issue at least twice but there must be something I'm missing, please help, I would be thankful!

-----
Nikola 

- [ Haven't gotten to that part yet] I have enabled debug mode
- [ Y] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Comment by snipe** [comment]
> sudo composer install --no-dev --prefer-source

Never run composer as root (or sudo). https://snipe-it.readme.io/docs/install-dependencies

Install composer globally using these instructions: https://getcomposer.org/doc/00-intro.md#globally

**Comment by Karnauday** [comment]
Hello,
I am trying to install Snipe-IT with WAMP on Windows Server 2012. I have followed most of the steps from [(https://github.com/snipe/snipe-it/issues/2530)]
But I am not able to generate a key using "php artisan key:generate".
Also when I tried putting this "composer create-project --prefer-dist laravel/laravel appName"

The key has been generated on command promt but didn't see any change on .env.

I don't know where i get struck and what i supposed to do now.  Can you please please help me out?
Will be highly grateful.

**Comment by snipe** [comment]
> Also when I tried putting this "composer create-project --prefer-dist laravel/laravel appName"

Don't do that. That's not what you need. That will create a new composer project.

You won't be able to generate an app key until you are able to run `composer install --no-dev --prefer-source`. What specific error are you getting?

(Also, next time maybe start your own issue, since this issue is related to LAMPP)

**Comment by Karnauday** [comment]
Hello Snipe,
I opened a new issue #3966.

Thank You

**Comment by Karnauday** [comment]
@Snipe,
I am not able to run ' composer install --no-dev --prefer-source'. When i put this:
Problem 1:
installation request for intervention/image 2.3. x-dev->satisfiable by intervention/image
intervention/image dev requires ext-fileinfo -> the requested php extension is missing from your system
What shall i do?
I am using WAMP/Windows Server 2012.
Thank You
